torchvision/prototype/transforms/_augment.py

```diff
@@ -34,6 +34,10 @@ def __init__(
         if p < 0 or p > 1:
             raise ValueError("Random erasing probability should be between 0 and 1")
         # TODO: deprecate p in favor of wrapping the transform in a RandomApply
+        # The above approach is very composable but will lead to verbose code. Instead, we can create a base class
+        # that inherits from Transform (say RandomTransform) that receives the `p` on constructor and by default
+        # implements the `p` random check on forward. This is likely to happen on the final clean ups, so perhaps
+        # update the comment to indicate accordingly OR create an issue to track this discussion.
```

Comment:
:+1:

Reply:
We need to decide here if we want BC or not. You can't adopt this proposal without offering a new API that achieves a minimum BC-level. Personally I don't favour the verbosity that this brings and how trivial some of the transforms become. For example what's the use of `HorizontalFlip` if it always flips? Why not use a lambda + the underlying kernel? You can argue that this would be even more composable and have even less classes to maintain.

Personally I think to get the maximum chances of rolling out the transforms successfully and getting adoption is to offer straight replacements of the existing classes and limited-time support for newly deprecated classes (with clear messaging on how to move out of them). Think of it as having a limited budget of breakages you can make. Would you spend it on RandomHorizontalFlip or on fixing the ToTensor and eliminating the use of ConvertImageDtype? I would definitely choose the latter.

Reply:
As discussed offline, even if we want this change, it has a low priority. I'm going to remove the comments.